python/cudf/cudf/core/column_accessor.py

```diff
@@ -270,16 +296,29 @@ def select_by_index(self, index: Any) -> ColumnAccessor:
             data, multiindex=self.multiindex, level_names=self.level_names,
         )
 
-    def set_by_label(self, key: Any, value: Any):
+    def set_by_label(self, key: Any, value: Any, validate: bool = True):
         """
         Add (or modify) column by name.
 
         Parameters
         ----------
-        key : name of the column
+        key
+            name of the column
         value : column-like
+            The value to insert into the column.
+        validate : bool
+            If True, the provided value will be coerced to a column and
+            validated before setting (Default value = True).
         """
         key = self._pad_key(key)
+        if validate:
+            value = column.as_column(value)
+            if len(self._data) > 0:
+                if len(value) != self._column_length:
+                    raise ValueError("All columns must be of equal length")
```

Comment:
Looks like we changed the error message here -- we should fix the corresponding test in https://github.com/rapidsai/cudf/blob/d500142e91c041a69b531d06f346e1630a8af046/python/cudf/cudf/tests/test_dataframe.py#L5225

Reply:
I think we changed the other error messages back to what they were

Reply:
I made this change yesterday, looks like I missed one. I think I got them all this time, tests are passing locally.